com.hazelcast.internal.server.tcp.LocalAddressRegistryIntegrationTest.whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up

_master_ (commit e8370b2db3b3ad114de71d6f5ff47a0751389b91)

Failed on Sonar build (Oracle JDK 11): https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK8-nightly/902/testReport/com.hazelcast.internal.server.tcp/LocalAddressRegistryIntegrationTest/whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: expected:<0> but was:<2>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.internal.server.tcp.LocalAddressRegistryIntegrationTest.lambda$whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up$3(LocalAddressRegistryIntegrationTest.java:160)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1226)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueEventually(HazelcastTestSupport.java:1243)
	at com.hazelcast.internal.server.tcp.LocalAddressRegistryIntegrationTest.whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up(LocalAddressRegistryIntegrationTest.java:159)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

</details>

<details><summary>Standard output:</summary>

```
02:41:35,892  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MetricsConfigHelper] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [LOCAL] [dev] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
02:41:35,893  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [logo] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
02:41:35,893  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
02:41:35,893  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220622 - e8370b2) starting at [127.0.0.1]:5701
02:41:35,893  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Cluster name: dev
02:41:35,893  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
02:41:35,893  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Jet is enabled
02:41:35,897  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MetricsConfigHelper] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
02:41:35,902  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Node] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Using TCP/IP discovery
02:41:35,902  WARN |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [CPSubsystem] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
02:41:35,904  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JetServiceBackend] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
02:41:35,905  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Diagnostics] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
02:41:35,905  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [LifecycleService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is STARTING
02:41:35,906  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [ClusterService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:1} [
	Member [127.0.0.1]:5701 - 49afd2ee-7a30-4628-9c54-70839edd621a this
]

02:41:35,906  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Jet started scanning for jobs
02:41:35,906 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:35,906  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [LifecycleService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is STARTED
02:41:35,907  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MetricsConfigHelper] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [LOCAL] [dev] [5.2-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
02:41:35,908  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [logo] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 
	+       +  o    o     o     o---o o----o o      o---o     o     o----o o--o--o
	+ +   + +  |    |    / \       /  |      |     /         / \    |         |   
	+ + + + +  o----o   o   o     o   o----o |    o         o   o   o----o    |   
	+ +   + +  |    |  /     \   /    |      |     \       /     \       |    |   
	+       +  o    o o       o o---o o----o o----o o---o o       o o----o    o   
02:41:35,908  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Copyright (c) 2008-2022, Hazelcast, Inc. All Rights Reserved.
02:41:35,908  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Hazelcast Platform 5.2-SNAPSHOT (20220622 - e8370b2) starting at [127.0.0.1]:5702
02:41:35,908  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Cluster name: dev
02:41:35,908  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Integrity Checker is disabled. Fail-fast on corrupted executables will not be performed. For more information, see the documentation for Integrity Checker.
02:41:35,908  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [system] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Jet is enabled
02:41:35,910  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MetricsConfigHelper] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
02:41:35,914  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Node] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Using TCP/IP discovery
02:41:35,914  WARN |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [CPSubsystem] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
02:41:35,915  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JetServiceBackend] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Setting number of cooperative threads and default parallelism to 2
02:41:35,917  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Diagnostics] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
02:41:35,917  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [LifecycleService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is STARTING
02:41:35,919  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.agitated_spence.IO.thread-out-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:32821 and /127.0.0.1:5701
02:41:35,919  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.goofy_spence.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5701 and /127.0.0.1:32821
02:41:36,006 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,107 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,207 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,307 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-3 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,407 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,508 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,608 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,708 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,808 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:36,908 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:37,008 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
02:41:37,020  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [ClusterService] hz.goofy_spence.generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 49afd2ee-7a30-4628-9c54-70839edd621a this
	Member [127.0.0.1]:5702 - 34fe115c-665c-458a-8e40-aa160b75e00d
]

02:41:37,020  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [ClusterService] hz.agitated_spence.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] 

Members {size:2, ver:2} [
	Member [127.0.0.1]:5701 - 49afd2ee-7a30-4628-9c54-70839edd621a
	Member [127.0.0.1]:5702 - 34fe115c-665c-458a-8e40-aa160b75e00d this
]

02:41:37,021  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Jet started scanning for jobs
02:41:37,021  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [LifecycleService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is STARTED
02:41:37,022  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [PartitionStateManager] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initializing cluster partition table arrangement...
02:41:37,024  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.goofy_spence.IO.thread-out-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:53131 and /127.0.0.1:5702
02:41:37,024  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.agitated_spence.IO.thread-in-1 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5702 and /127.0.0.1:53131
02:41:37,109  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 0, bind-any: true
02:41:37,109  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.goofy_spence.IO.thread-out-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:36219 and /127.0.0.1:5702
02:41:37,109  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.agitated_spence.IO.thread-in-2 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5702 and /127.0.0.1:36219
02:41:37,110  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.agitated_spence.IO.thread-out-0 - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:54033 and /127.0.0.1:5701
02:41:37,110  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.goofy_spence.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Initialized new cluster connection between /127.0.0.1:5701 and /127.0.0.1:54033
02:41:37,128  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [LifecycleService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
02:41:37,129 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Added a shutting-down member: 34fe115c-665c-458a-8e40-aa160b75e00d
02:41:37,130  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MigrationManager] hz.goofy_spence.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5702 - 34fe115c-665c-458a-8e40-aa160b75e00d is handled
02:41:37,130  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MigrationManager] hz.goofy_spence.migration - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Repartitioning cluster data. Migration tasks count: 6
02:41:37,133  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MigrationManager] hz.goofy_spence.migration - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] All migration tasks have been completed. (repartitionTime=Wed Jun 22 02:41:37 UTC 2022, plannedMigrations=6, completedMigrations=6, remainingMigrations=0, totalCompletedMigrations=6, elapsedMigrationOperationTime=14ms, totalElapsedMigrationOperationTime=14ms, elapsedDestinationCommitTime=0ms, totalElapsedDestinationCommitTime=0ms, elapsedMigrationTime=16ms, totalElapsedMigrationTime=16ms)
02:41:37,134  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Node] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Shutting down connection manager...
02:41:37,134  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=1, /127.0.0.1:32821->/127.0.0.1:5701, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5701, remoteUuid=49afd2ee-7a30-4628-9c54-70839edd621a, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: TcpServer is stopping
02:41:37,134  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=2, /127.0.0.1:5702->/127.0.0.1:53131, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5701, remoteUuid=49afd2ee-7a30-4628-9c54-70839edd621a, alive=false, connectionType=MEMBER, planeIndex=1] closed. Reason: TcpServer is stopping
02:41:37,134  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=4, /127.0.0.1:54033->/127.0.0.1:5701, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5701, remoteUuid=49afd2ee-7a30-4628-9c54-70839edd621a, alive=false, connectionType=MEMBER, planeIndex=2] closed. Reason: TcpServer is stopping
02:41:37,134  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Connection[id=3, /127.0.0.1:5702->/127.0.0.1:36219, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5701, remoteUuid=49afd2ee-7a30-4628-9c54-70839edd621a, alive=false, connectionType=MEMBER, planeIndex=2] closed. Reason: TcpServer is stopping
02:41:37,134  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Node] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Shutting down node engine...
02:41:37,134  WARN |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.goofy_spence.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=1, /127.0.0.1:5701->/127.0.0.1:32821, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5702, remoteUuid=34fe115c-665c-458a-8e40-aa160b75e00d, alive=false, connectionType=MEMBER, planeIndex=0] closed. Reason: Exception in Connection[id=1, /127.0.0.1:5701->/127.0.0.1:32821, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5702, remoteUuid=34fe115c-665c-458a-8e40-aa160b75e00d, alive=true, connectionType=MEMBER, planeIndex=0], thread=hz.goofy_spence.IO.thread-in-0
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_311]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_311]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_311]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_311]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:378) ~[?:1.8.0_311]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:115) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
02:41:37,134  WARN |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.goofy_spence.IO.thread-in-2 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=3, /127.0.0.1:36219->/127.0.0.1:5702, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5702, remoteUuid=34fe115c-665c-458a-8e40-aa160b75e00d, alive=false, connectionType=MEMBER, planeIndex=2] closed. Reason: Exception in Connection[id=3, /127.0.0.1:36219->/127.0.0.1:5702, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5702, remoteUuid=34fe115c-665c-458a-8e40-aa160b75e00d, alive=true, connectionType=MEMBER, planeIndex=2], thread=hz.goofy_spence.IO.thread-in-2
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_311]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_311]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_311]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_311]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:378) ~[?:1.8.0_311]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:115) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
02:41:37,134  WARN |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.goofy_spence.IO.thread-in-1 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=2, /127.0.0.1:53131->/127.0.0.1:5702, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5702, remoteUuid=34fe115c-665c-458a-8e40-aa160b75e00d, alive=false, connectionType=MEMBER, planeIndex=1] closed. Reason: Exception in Connection[id=2, /127.0.0.1:53131->/127.0.0.1:5702, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5702, remoteUuid=34fe115c-665c-458a-8e40-aa160b75e00d, alive=true, connectionType=MEMBER, planeIndex=1], thread=hz.goofy_spence.IO.thread-in-1
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_311]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_311]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_311]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_311]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:378) ~[?:1.8.0_311]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:115) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
02:41:37,136  WARN |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnection] hz.goofy_spence.IO.thread-in-0 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connection[id=4, /127.0.0.1:5701->/127.0.0.1:54033, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5702, remoteUuid=34fe115c-665c-458a-8e40-aa160b75e00d, alive=false, connectionType=MEMBER, planeIndex=2] closed. Reason: Exception in Connection[id=4, /127.0.0.1:5701->/127.0.0.1:54033, qualifier=EndpointQualifier{type='MEMBER'}, endpoint=[127.0.0.1]:5702, remoteUuid=34fe115c-665c-458a-8e40-aa160b75e00d, alive=true, connectionType=MEMBER, planeIndex=2], thread=hz.goofy_spence.IO.thread-in-0
java.io.IOException: Connection reset by peer
	at sun.nio.ch.FileDispatcherImpl.read0(Native Method) ~[?:1.8.0_311]
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:39) ~[?:1.8.0_311]
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223) ~[?:1.8.0_311]
	at sun.nio.ch.IOUtil.read(IOUtil.java:197) ~[?:1.8.0_311]
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:378) ~[?:1.8.0_311]
	at com.hazelcast.internal.networking.nio.NioInboundPipeline.process(NioInboundPipeline.java:115) ~[classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKey(NioThread.java:383) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.processSelectionKeys(NioThread.java:368) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.selectLoop(NioThread.java:294) [classes/:?]
	at com.hazelcast.internal.networking.nio.NioThread.executeRun(NioThread.java:249) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
02:41:37,137  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [NodeExtension] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Destroying node NodeExtension.
02:41:37,137  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Node] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 8 ms.
02:41:37,137  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [LifecycleService] whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up - [127.0.0.1]:5702 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
02:41:37,137  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 0, bind-any: true
02:41:37,137  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Could not connect to: /127.0.0.1:5702. Reason: IOException[Connection refused to address /127.0.0.1:5702]
02:41:37,238  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 0, bind-any: true
02:41:37,238  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Could not connect to: /127.0.0.1:5702. Reason: IOException[Connection refused to address /127.0.0.1:5702]
02:41:37,310  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 0, bind-any: true
02:41:37,310  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Could not connect to: /127.0.0.1:5702. Reason: IOException[Connection refused to address /127.0.0.1:5702]
02:41:37,338  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 0, bind-any: true
02:41:37,339  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Could not connect to: /127.0.0.1:5702. Reason: IOException[Connection refused to address /127.0.0.1:5702]
02:41:37,410  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 0, bind-any: true
02:41:37,411  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Could not connect to: /127.0.0.1:5702. Reason: IOException[Connection refused to address /127.0.0.1:5702]
02:41:37,439  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Connecting to /127.0.0.1:5702, timeout: 0, bind-any: true
02:41:37,439  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnector] hz.goofy_spence.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Could not connect to: /127.0.0.1:5702. Reason: IOException[Connection refused to address /127.0.0.1:5702]
02:41:37,440  WARN |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TcpServerConnectionErrorHandler] hz.goofy_spence.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Removing connection to endpoint [127.0.0.1]:5702 Cause => java.io.IOException {Connection refused to address /127.0.0.1:5702}, Error-Count: 5
02:41:37,440  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MembershipManager] hz.goofy_spence.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Removing Member [127.0.0.1]:5702 - 34fe115c-665c-458a-8e40-aa160b75e00d
02:41:37,440  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [ClusterService] hz.goofy_spence.cached.thread-6 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] 

Members {size:1, ver:3} [
	Member [127.0.0.1]:5701 - 49afd2ee-7a30-4628-9c54-70839edd621a this
]

02:41:37,440  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [TransactionManagerService] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Committing/rolling-back live transactions of [127.0.0.1]:5702, UUID: 34fe115c-665c-458a-8e40-aa160b75e00d
02:41:37,440 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partition replication is not in safe state, but in REPLICA_NOT_SYNC
02:41:37,440 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-7 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Removed a shutting-down member: 34fe115c-665c-458a-8e40-aa160b75e00d, now shuttingDownMembers=[]
02:41:37,440 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] hz.goofy_spence.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Not starting jobs because partition replication is not in safe state, but in MIGRATION_LOCAL
02:41:37,441  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MigrationManager] hz.goofy_spence.migration - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Partition balance is ok, no need to repartition.
02:41:37,540 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobRepository] hz.goofy_spence.cached.thread-8 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Job cleanup took 0ms
02:41:42,540 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobRepository] hz.goofy_spence.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Job cleanup took 0ms
02:41:47,558 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobRepository] hz.goofy_spence.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Job cleanup took 0ms
02:41:47,577  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [LifecycleService] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
02:41:47,577 DEBUG |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [JobCoordinationService] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Added a shutting-down member: 49afd2ee-7a30-4628-9c54-70839edd621a
02:41:47,577  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [MigrationManager] hz.goofy_spence.cached.thread-5 - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutdown request of Member [127.0.0.1]:5701 - 49afd2ee-7a30-4628-9c54-70839edd621a this is handled
02:41:47,578  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Node] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutting down connection manager...
02:41:47,579  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Node] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Shutting down node engine...
02:41:47,582  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [NodeExtension] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Destroying node NodeExtension.
02:41:47,582  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [Node] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Hazelcast Shutdown is completed in 5 ms.
02:41:47,582  INFO |whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up| - [LifecycleService] main - [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up(com.hazelcast.internal.server.tcp.LocalAddressRegistryIntegrationTest): BuildInfo{version='5.2-SNAPSHOT', build='20220622', buildNumber=20220622, revision=e8370b2, enterprise=false, serializationVersion=1}
Hiccups measured while running test 'whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up(com.hazelcast.internal.server.tcp.LocalAddressRegistryIntegrationTest):'
02:41:35, accumulated pauses: 50 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:41:40, accumulated pauses: 50 ms, max pause: 0 ms, pauses over 1000 ms: 0
02:41:45, accumulated pauses: 428 ms, max pause: 406 ms, pauses over 1000 ms: 0
```

</details>

--- a/hazelcast/src/test/java/com/hazelcast/internal/server/tcp/LocalAddressRegistryIntegrationTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/internal/server/tcp/LocalAddressRegistryIntegrationTest.java
@@ -25,13 +25,14 @@ import com.hazelcast.core.Hazelcast;
 import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.instance.EndpointQualifier;
 import com.hazelcast.instance.impl.Node;
-import com.hazelcast.internal.nio.Connection;
+import com.hazelcast.internal.server.ServerConnection;
 import com.hazelcast.map.IMap;
 import com.hazelcast.partition.Partition;
 import com.hazelcast.partition.PartitionService;
 import com.hazelcast.test.HazelcastSerialClassRunner;
 import com.hazelcast.test.HazelcastTestSupport;
 import com.hazelcast.test.annotation.NightlyTest;
+
 import org.junit.After;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
@@ -43,7 +44,11 @@ import java.util.Random;
 import java.util.UUID;
 
 import static com.hazelcast.internal.util.ExceptionUtil.rethrow;
+import static com.hazelcast.spi.properties.ClusterProperty.CHANNEL_COUNT;
 import static com.hazelcast.test.Accessors.getNode;
+import static org.hamcrest.CoreMatchers.hasItem;
+import static org.hamcrest.CoreMatchers.not;
+import static org.hamcrest.MatcherAssert.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
@@ -51,6 +56,8 @@ import static org.junit.Assert.assertNull;
 @RunWith(HazelcastSerialClassRunner.class)
 @Category(NightlyTest.class)
 public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
+
+    private static final int CHANNELS_PER_CONNECTION = 3;
 
     // MEMBER & WAN & CLIENT addresses of the connection initiator
     private static final Address INITIATOR_MEMBER_ADDRESS;
@@ -84,8 +91,6 @@ public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
     @Test
     public void whenSomeConnectionClosedBetweenMembers_registeredAddresses_should_not_be_cleaned_up() {
         int timeoutSecs = 10;
-        int tcpChannelsPerConnection = 3;
-        System.setProperty("tcp.channels.per.connection", String.valueOf(tcpChannelsPerConnection));
 
         // create members
         HazelcastInstance serverMember = Hazelcast.newHazelcastInstance(createConfigForServer());
@@ -101,7 +106,7 @@ public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
                 assertGreaterOrEquals(
                         "The number of connections must be greater than the number of channels.",
                         connectionManager.getConnections().size(),
-                        tcpChannelsPerConnection
+                        CHANNELS_PER_CONNECTION
                 ), timeoutSecs);
 
         LinkedAddresses registeredAddressesOfInitiatorMember = serverNode
@@ -109,13 +114,9 @@ public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
                 .linkedAddressesOf(initiatorMemberUuid);
         assertNotNull(registeredAddressesOfInitiatorMember);
         assertContains(registeredAddressesOfInitiatorMember.getAllAddresses(), INITIATOR_MEMBER_ADDRESS);
-        int previousNoConnections = connectionManager.getConnections().size();
-        closeRandomConnection(new ArrayList<>(connectionManager.getConnections()));
-        assertTrueEventually(() ->
-                assertEquals(
-                        previousNoConnections - 1,
-                        connectionManager.getConnections().size()
-                ), timeoutSecs);
+        ServerConnection removedConnection = closeRandomConnection(new ArrayList<>(connectionManager.getConnections()));
+        assertTrueEventually(() -> assertThat(connectionManager.getConnections(), not(hasItem(removedConnection))),
+                timeoutSecs);
 
         LinkedAddresses registeredAddressesAfterConnectionClose = serverNode
                 .getLocalAddressRegistry()
@@ -129,8 +130,6 @@ public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
     @Test
     public void whenAllConnectionClosedBetweenMembers_registeredAddresses_should_be_cleaned_up() {
         int timeoutSecs = 10;
-        int tcpChannelsPerConnection = 3;
-        System.setProperty("tcp.channels.per.connection", String.valueOf(tcpChannelsPerConnection));
 
         // create members
         HazelcastInstance serverMember = Hazelcast.newHazelcastInstance(createConfigForServer());
@@ -146,7 +145,7 @@ public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
                 assertGreaterOrEquals(
                         "The number of connections must be greater than the number of channels.",
                         connectionManager.getConnections().size(),
-                        tcpChannelsPerConnection
+                        CHANNELS_PER_CONNECTION
                 ), timeoutSecs);
 
         LinkedAddresses registeredAddressesOfInitiatorMember = serverNode
@@ -170,7 +169,7 @@ public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
 
 
     private Config createConfigForServer() {
-        Config config = smallInstanceConfig();
+        Config config = smallInstanceConfig().setProperty(CHANNEL_COUNT.getName(), String.valueOf(CHANNELS_PER_CONNECTION));
         AdvancedNetworkConfig advancedNetworkConfig = config.getAdvancedNetworkConfig();
         JoinConfig advancedJoinConfig = advancedNetworkConfig.getJoin();
         advancedJoinConfig.getTcpIpConfig().setEnabled(true);
@@ -194,7 +193,7 @@ public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
     }
 
     private Config createConfigForInitiator() {
-        Config config = smallInstanceConfig();
+        Config config = smallInstanceConfig().setProperty(CHANNEL_COUNT.getName(), String.valueOf(CHANNELS_PER_CONNECTION));
         AdvancedNetworkConfig advancedNetworkConfig = config.getAdvancedNetworkConfig();
         JoinConfig advancedJoinConfig = advancedNetworkConfig.getJoin();
         advancedJoinConfig.getTcpIpConfig().setEnabled(true).addMember(
@@ -231,10 +230,12 @@ public class LocalAddressRegistryIntegrationTest extends HazelcastTestSupport {
         dummy.destroy();
     }
 
-    private void closeRandomConnection(List<Connection> connections) {
+    private ServerConnection closeRandomConnection(List<ServerConnection> connections) {
         Random random = new Random();
         int randomIdx = random.nextInt(connections.size());
-        connections.get(randomIdx).close("Failure is injected", null);
+        ServerConnection connection = connections.get(randomIdx);
+        connection.close("Failure is injected", null);
+        return connection;
     }
 
     private String randomKeyNameOwnedByPartition(HazelcastInstance hz, int partitionId) {

